### PR TITLE
NDI-9: Fix table and spacing glitches in event pages

### PR DESCRIPTION
**Pull request by mukeshcompucorp**
# Description
This PR
- Fixes minor style glitches around event pages to make it look better.
- Adds new function `civi-dialog` to generate selector for dialog.

# Screenshots
**Before calendar Border and padding**
image
**After**
image

**Before Date Range spacing**
image
**After**
image

**Before excessive shadows**
image
**After**
image

**Before Address tab**
before-address
**After**
after-address

**Before Fees tab**
before-fees
**After**
after-fees

**Before Tell a friend tab**
before-friend

**After**
after-friend

# Technical Details
New function `civi-dialog` is added to generate selector for dialog for civicrm.
```scss
@function civi-dialog() {
  @return unquote('.crm-container[role=\'dialog\']');
}
```
# Notes
File `scss/bootstrap/overrides/style/_ui-select.scss` is not stylelint verified for all the issues, below issues are skipped.
```bash
15:21  ✖  Unexpected qualifying type selector   selector-no-qualifying-type
89:34  ✖  Unexpected qualifying type selector   selector-no-qualifying-type
```
Which is about using selector as tag + class combination,
```scss
input.ui-select-search {
  height: 1.9em;
  text-indent: 10px;
}
```
```scss
a.btn {
   margin-right: 14px !important;
}
```
# Comments
- BackstopJS tested
- Stylelint tested

### CODE REVIEW

File: scss/civicrm/contact/_tabs.scss [hunk 1]
```diff
@@ -11,11 +13,11 @@
       border-radius: 0;
       box-shadow: $box-shadow;
       float: left;
-      padding: 0 0 99999px;
       margin-bottom: -99999px;
+      padding: 0 0 99999px;
```
Comment (AkA84) [inline scss/civicrm/contact/_tabs.scss]: why this value?

File: scss/civicrm/contact/_tabs.scss [hunk 2]
```diff
@@ -11,11 +13,11 @@
...
       width: 200px;
 
-      &~.ui-tabs-panel {
+      ~.ui-tabs-panel {
```
Comment (AkA84) [inline scss/civicrm/contact/_tabs.scss]: so was this selector wrong all along?

File: scss/civicrm/contact/_tabs.scss [hunk 3]
```diff
@@ -178,15 +180,11 @@
   /* override of jquery-ui.css that creates weird white background */
   .ui-widget-content {
     background: transparent;
+    border: 0 !important;
+    color: #222;
```
Comment (AkA84) [inline scss/civicrm/contact/_tabs.scss]: what's this color? Shouldn't we have it in the palette?

File: scss/civicrm/event/_detail.scss [hunk 4]
```diff
@@ -174,3 +162,22 @@
   margin-top: 20px !important;
 }
 
+
+.crm-container {
```
Comment (AkA84) [inline scss/civicrm/event/_detail.scss]: why is `.crm-container` necessary?

File: scss/bootstrap/overrides/style/_ui-select.scss [hunk 5]
```diff
@@ -1,7 +1,5 @@
-.ui-select-bootstrap[theme="civihr-ui-select"] {
-
+.ui-select-bootstrap[theme='civihr-ui-select'] {
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_ui-select.scss]: this is a bootstrap/ partial, what does it have to do with core screens?

File: scss/civicrm/event/_detail.scss (hunk 4 above)
Comment (mukeshcompucorp) [reply scss/civicrm/event/_detail.scss]: This specific style is for popup where `#crm-container` does not exist so, we need `.crm-container`.

File: scss/bootstrap/overrides/style/_ui-select.scss (hunk 5 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/style/_ui-select.scss]: This is stylelint change, nothing to do with core screen.

File: scss/civicrm/contact/_tabs.scss (hunk 1 above)
Comment (mukeshcompucorp) [reply scss/civicrm/contact/_tabs.scss]: I haven't changed any on this file, this is stylelint change, so, I have just added header of `stylelint-disable`

File: scss/civicrm/contact/_tabs.scss (hunk 2 above)
Comment (mukeshcompucorp) [reply scss/civicrm/contact/_tabs.scss]: Yeah, we actually don't need & sign for such selection, & is generally needed when we want to combine multiple classes,
like
```scss
.class1 {
  &.class2 {
    //style here
  }
}
```

The above one with have same output weather we use & or not. (Though, this is stylelint change nothing to do with current ticket)

File: scss/civicrm/event/_detail.scss (hunk 4 above)
Comment (AkA84) [reply scss/civicrm/event/_detail.scss]: sorry @mukeshcompucorp but you need to provide more details here. Why are you talking about a popup, if the selectors don't mention any popup? Why would `#crm-container` be missing? Can you show some screenshots of what you are talking about?

Comment (mukeshcompucorp) [reply scss/civicrm/event/_detail.scss]: The style we are handling here is for below popup dialog.
image

Dialog is having `.crm-container` as wrapper to our form.

File: scss/civicrm/contact/_tabs.scss (hunk 3 above)
Comment (mukeshcompucorp) [reply scss/civicrm/contact/_tabs.scss]: This again is stylelint change. I know that, this is not our palette color but, currently I have not modified anything to it to keep my scope limited to actual task.

**Pull request merged**